apps/src/code-studio/pd/teacher_application/teacher_application.jsx

```diff
@@ -0,0 +1,374 @@
+        {this.buttonList('check', 'What grades are you teaching in the current 2017-18 school year? (select all that apply)', 'grades2017', grades)}
+        {this.buttonList('check', 'What subjects are you teaching in the current 2016-17 school year? (select all that apply)', 'subjects2017', subjects, true)}
+        <FieldGroup
+          id="principalFirstName"
+          label="Principal's first name"
+          type="text"
+        />
+        <FieldGroup
+          id="principalLastName"
+          label="Principal's last name"
+          type="text"
+        />
+        <FormGroup>
+          <ControlLabel>
+            Principal's prefix
+          </ControlLabel>
+          <FormControl componentClass="select" id="principalPrefix">
+            <option value="Dr.">Dr.</option>
+            <option value="Miss.">Miss.</option>
+            <option value="Mr.">Mr.</option>
+            <option value="Mrs.">Mrs.</option>
+            <option value="Ms.">Ms.</option>
+          </FormControl>
+        </FormGroup>
+        <FieldGroup
+          id="principalEmail"
+          label="Principal's Email Address"
+          type="text"
+        />
+      </div>
+    );
+  },
+
+  renderCourseSelection() {
+    return (
+      <div>
+        <ControlLabel>
+          Which professional learning program are you applying to join for the 2017-18 school year? Click on each
+          curriculum for more information. Note: this application is only for Computer Science Discoveries and Computer
+          Science Principles. If you are interested in <a href="">Computer Science Fundamentals</a>, please
+          visit <a href="">this</a> page for information about workshops in your area. (Select one)
+        </ControlLabel>
+        <Radio
+          value="csd"
+          name="courseSelection"
+          onChange={this.handleCourseChange}
+        >
+          <a href="">Computer Science Discoveries</a> (designed for 7th - 9th grade)
+        </Radio>
+        <Radio
+          value="csp"
+          name="courseSelection"
+          onChange={this.handleCourseChange}
+        >
+          <a href="">Computer Science Principles</a> (designed for 9th - 12th grade, and can be implemented as an AP or introductory course)
+        </Radio>
+      </div>
+    );
+  },
+
+  handleCourseChange(event) {
+    this.setState({selectedCourse: event.target.value});
+  },
+
+  renderCSDSpecificContent() {
+    if (this.state.selectedCourse === 'csd') {
+      return (
+        <div id="csdSpecificContent">
+          {this.buttonList(
+            'check',
+            'To which grades do you plan to teach CS Discoveries? Please note that the CS Discoveries Professional ' +
+            'Learning Program is not available for grades K-5. (select all that apply)',
+            'csdGrades',
+            _.range(6,13)
+          )}
+        </div>
+      );
+    }
+  },
+
+  renderCSPSpecificContent() {
```

Comment:
these render*SpecificContent functions could also be stored in the above COURSES, i.e. a property called `renderContent: ` that points to the appropriate function.